### ISSUE TITLE
Error 404 on gitterAPI call

### ISSUE THREAD

**Issue by chaostools** [coding-blocks/Chatter#42]
At first after adding logs to all requests, I found that the server is not giving any response (404 not found) due to which the JSONObject Class is unable to create a json object
- The shared preferences are successfully saving the user's access token
- the okhttp request 
```java 
Request request = new Request.Builder()
                        .url("https://gitter.im/v1/user/me")
                        .addHeader("Accept", "application/json")
                        .addHeader("Authorization:", "Bearer " + accessToken)
                        .build();
```
is giving no response(404)
- thus the `JSONObject` is not created.

Here are the detailed logs:
```
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: onCreateView:=====================WELCOME TO THE ROOM(S) FRAGMENT!!=================
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: onCreateView: Realm initialized and got its instance.querying for results
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: onCreateView: realm query result (rooms):[]
onCreateView: adding on change listener for live results on realmResults (room) 
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: onCreateView: setting rv layout manager and  calling display rooms for the following result:[]
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: displayRooms: called with realmResult(rooms):[]
displayRooms: roomsize=0, calling getrooms(1)
getRooms: called with severity=1
isNetworkAvailable: called
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: network is availabale
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: getting access token from shared prefs->UserPreferences
getRooms: token recievede887b3aab4a137af403da8e65ebd9f63039d60fa
getRooms: creating OkHTTP Request
6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: request created:Request{method=GET, url=https://gitter.im/v1/rooms, tag=null}
getRooms:  creating new call using request and enqueue
.301 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: displayRooms: recycle view setted with  rooms:[]
displayRooms: calling getrooms(severity:0)
getRooms: called with severity=0
isNetworkAvailable: called
.302 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: network is availabale
.308 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: getting access token from shared prefs->UserPreferences
getRooms: token recieved :######my_access_token##################
getRooms: creating OkHTTP Request
.309 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: getRooms: request created:Request{method=GET, url=https://gitter.im/v1/rooms, tag=null}
getRooms:  creating new call using request and enqueue
.811 6554-6604/com.codingblocks.chatter E/NativeCrypto: ssl=0xb8f9e0a0 cert_verify_callback x509_store_ctx=0xa1b10460 arg=0x0
ssl=0xb8f9e0a0 cert_verify_callback calling verifyCertificateChain authMethod=ECDHE_RSA
.937 6554-6606/com.codingblocks.chatter E/NativeCrypto: ssl=0xb9096c90 cert_verify_callback x509_store_ctx=0xa1a0c460 arg=0x0
ssl=0xb9096c90 cert_verify_callback calling verifyCertificateChain authMethod=ECDHE_RSA
30.389 6554-6604/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: client:onResponse:response recieved: Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}
client:onResponse:responseText:{"rooms":Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}}
client:onResponse:started running on UI Thread
30.403 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: runUI: inside run
30.404 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: runUI: trying to convert to json object
30.406 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: run: json exception occurred:Unterminated object at character 18 of {"rooms":Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}}
30.589 6554-6606/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: client:onResponse:response recieved: Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}
client:onResponse:responseText:{"rooms":Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}}
client:onResponse:started running on UI Thread
30.590 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: runUI: inside run
runUI: trying to convert to json object
30.595 6554-6554/com.codingblocks.chatter E/ANSH>>ROOMFRAG>>: run: json exception occurred:Unterminated object at character 18 of {"rooms":Response{protocol=http/1.1, code=404, message=Not Found, url=https://gitter.im/v1/rooms}}


    

```

@the-dagger  @championswimmer

**Comment by abishekvashok** [comment]
Duplicate of #39

**Comment by chaostools** [comment]
@abishekvashok  can you please tell me how is this a duplicate? As far as I can see, I never saw that the app getting  'crashed' by the json request you wrote.Can you please add steps to reproduce your bug?

I have got a working project in my system and it was crashing due to wrong butterknife usage(But already fixed that)

Moreover ,  i think its the  api call that is giving a null response and not the json object builder.

**Comment by abishekvashok** [comment]
It's just another narrative bro, chill :)

**Comment by abishekvashok** [comment]
This is a duplicate of #39 : @championswimmer @the-dagger solved in #142
